src/compile/legend/encode.ts

```diff
@@ -66,9 +66,9 @@ export function symbols(fieldDef: FieldDef<string>, symbolsSpec: any, model: Uni
   }
 
   if (channel !== OPACITY) {
-    const opacityDef = model.encoding.opacity;
-    if (isValueDef(opacityDef)) {
-      symbols.opacity = {value: opacityDef.value};
+    const opacity = getOpacityValue(model.encoding.opacity);
+    if (opacity) { // only apply opacity if it is zero or undefined
```

Comment:
"if it is neither zero or undefined"

I think this is a typo that I made earlier :P